app/model/auth-message/auth-message.go

```diff
@@ -154,6 +192,46 @@ func generateFungibleArguments() (abi.Arguments, error) {
 		}}, nil
 }
 
+func generateFungibleArgumentsWithFee() (abi.Arguments, error) {
```

Comment:
This can reuse the `func generateFungibleArguments()` by appending the last argument for the fee to the result.